src/features/scheduling/generation/FreeplayScheduler.ts

```
@@ -1,42 +1,139 @@
+      });
+    }
+  }
 
-  withCampers(campers: CamperAttendee[]): FreeplayScheduler { this.campers = campers; return this; }
+  const camperIds = campers.map(camper => camper.attendeeId);
+  const employeeIds = [...staff, ...admins].map(employee => employee.attendeeId);
+  const buddyCandidatesById: { [attendeeId: number]: Set<number>; } = {};
+  attendees.forEach(attendee => buddyCandidatesById[attendee.attendeeId] = new Set((attendee.role === "CAMPER" ? employeeIds : camperIds).filter(potentialCandidateId => !attendee.snapshot.nonoList.includes(potentialCandidateId) && !buddiesInOtherFreeplays[attendee.attendeeId].has(potentialCandidateId) && (attendee.snapshot.gender !== "Female" || employeesById[potentialCandidateId].snapshot.gender === "Female"))));
 
-  withStaff(staff: StaffAttendee[]): FreeplayScheduler { this.staff = staff; return this; }
+  let unassignedCamperIds: number[] = camperIds;
```

ESLint

[error] 'unassignedCamperIds' is never reassigned. Use 'const' instead.

(prefer-const)